clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7834,6 +7834,9 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
     TranslatorArgs.push_back("-spirv-max-version=1.1");
     TranslatorArgs.push_back("-spirv-debug-info-version=legacy");
+    // Prevent crash in the translator if input IR contains DIExpression
+    // operations which don't have mapping to OpenCL.DebugInfo.100 spec.
+    TranslatorArgs.push_back("-spirv-allow-extra-diexpressions");
```

Comment:
I'd suggest safeguarding and adding similar code to `ToolChains/SYCL.cpp:SYCL::Linker::constructLLVMSpirvCommand`. @mdtoguchi, do you recall why it ended up with 2 similar Translator command creations? Was this part of @Fznamznon's effort on device code split?

Reply:
I have added this option to `ToolChains/SYCL.cpp:SYCL::Linker::constructLLVMSpirvCommand`.